lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/ContextStopExceptionTest.cs

```diff
@@ -0,0 +1,248 @@
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().SetNumber(2).Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                lock (_lock)
+                {
+                    if (_shouldSubmitFailEvaluatorContext)
+                    {
+                        // This context should fail the Evaluator upon calling ActiveContext.Dispose().
+                        value.SubmitContext(
+                            GetContextStopExceptionContextConfiguration(FailEvaluatorContextId));
+                        _shouldSubmitFailEvaluatorContext = false;
+                    }
+                    else
+                    {
+                        // This is the Context that will be stacked upon by ContextId1.
+                        value.SubmitContext(
+                            ContextConfiguration.ConfigurationModule
+                                .Set(ContextConfiguration.Identifier, ContextId0)
+                                .Build());
+                    }
+                }
+            }
+
+            /// <summary>
+            /// This will be ContextId1.
+            /// It will submit a Task to the parent of Context with ContextId1.
+            /// </summary>
+            public void OnNext(IFailedContext value)
+            {
+                Assert.Equal(ContextId1, value.Id);
+                Assert.True(value.ParentContext.IsPresent());
+                Assert.Equal(ContextId0, value.ParentContext.Value.Id);
+
+                // TODO[JIRA REEF-1468]: Validate that Exception is properly serialized.
+                Logger.Log(Level.Info, FailedContextReceived);
+
+                value.ParentContext.Value.SubmitTask(
+                    TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, TaskId)
+                        .Set(TaskConfiguration.Task, GenericType<NullTask>.Class)
+                        .Build());
+            }
+
+            /// <summary>
+            /// This will be a FailedEvaluator generated by the Context with ID as FailEvaluatorContextId.
+            /// </summary>
+            public void OnNext(IFailedEvaluator value)
+            {
+                Assert.Equal(1, value.FailedContexts.Count);
+                Assert.Equal(FailEvaluatorContextId, value.FailedContexts.First().Id);
+                Assert.NotNull(value.EvaluatorException.InnerException);
+                Assert.True(value.EvaluatorException.InnerException is TestSerializableException);
+                Assert.Equal(ExpectedException, value.EvaluatorException.InnerException.Message);
+                Logger.Log(Level.Info, FailedEvaluatorReceived);
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                if (value.Id.Equals(FailEvaluatorContextId))
```

Comment:
If the context with FailEvaluatorContextId results in IFailedEvaluator, would the driver still receive IActiveContext event for it?